### ISSUE TITLE
Cppcheck-htmlreport doesn't work

### ISSUE THREAD

**Issue by mayl** [NixOS/nixpkgs#111303]
**Description**
running `cppcheck-htmlreport` which is a python script included with `cppcheck` fails because the `python` it references does not have `pygments` installed.

**To Reproduce**
1. `$ nix-shell -p cppcheck`
2. `$ cppcheck-htmlreport`
3. Observe error `Import Error: No module named pygments`

**Expected behavior**
Should run

**Notify maintainers**
@joachifm 
@peti 

**Metadata**
```
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.86, NixOS, 20.09.2497.4a75ca4a4e7 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.9`
 - channels(root): `"nixos-20.09.2497.4a75ca4a4e7, nixos-unstable-21.03pre261529.d9dba88d08a"`
 - channels(larry): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by mayl** [comment]
The Shebang at the top of `cppcheck-htmlreport` has a path to a python interperter in the `nix-store`, but looking at the `nix-expression` I don't know how it gets there.  Can someone explain how it gets in there?

**Comment by mayl** [comment]
Not ready to write a pull request yet, but documenting for when I have time to circle back and fix it that I was able to get this working with an override that sets ```depsHostHost = [pkgs.python.withPackages(ps: [ps.pygments]) ```

**Comment by aanderse** [comment]
I'm also curious **exactly** why `python` is being picked up. Taking a **very** quick look over the derivation I can only assume `python` is being picked up by one of the other inputs, but I really would like to understand concretely why.

**Comment by mayl** [comment]
Yes, me too!  As best as I can tell, `mkDerivation` automatically runs [`patch-shebangs.sh`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/patch-shebangs.sh) which is why the shebang in `cppcheck-htmlreport` gets re-written, but how python enters the derivation exactly is not clear.
